### PR TITLE
[openpay] Create an OpenPay benchmark

### PR DESCRIPTION
**Pull request by iskakaushik**
Adds support for running openpay benchmarks on framework postsubmits.

### CODE REVIEW

File: .ci.yaml [hunk 1]
```diff
@@ -2345,6 +2345,18 @@ targets:
       task_name: android_lifecycles_test
     scheduler: luci
 
+  - name: Linux_android openpay_benchmarks__scroll_perf
+    bringup: true
+    recipe: devicelab/devicelab_drone
+    presubmit: false
+    timeout: 60
+    properties:
+      openpay: true
```
Comment (iskakaushik) [inline .ci.yaml:2354]: Looks like this is not the right way to add the properties.  Is there a pointer to the right way? or do we need to fix the CI.yaml validation check?

File: dev/devicelab/lib/framework/utils.dart [hunk 2]
```diff
@@ -585,6 +585,9 @@ void cd(dynamic directory) {
 
 Directory get flutterDirectory => Directory.current.parent.parent;
 
+// TODO (kaushik) : this is currently not discoverable, can we set an env var?
+Directory get openpayDirectory => Directory.current.parent.parent;
```
Comment (iskakaushik) [inline dev/devicelab/lib/framework/utils.dart:589]: Would it be possible to have an environment variable like `OPENPAY_CHECKOUT_PATH` be passed? Looking [here](https://flutter-review.googlesource.com/c/recipes/+/31583/4/recipes/devicelab/devicelab_drone.py#65) the checkout seems to be at a temp path.

**Review by iskakaushik — COMMENTED** [review]
@keyonghan I have a few questions added here, could you please provide some guidance on those?

I was also wondering if there was anything else that needs to be done to have the benchmark results be posted to skia perf. 

Thanks!

cc: @jonahwilliams , @dnfield

File: dev/devicelab/lib/framework/utils.dart (hunk 2 above)
Comment (keyonghan) [reply dev/devicelab/lib/framework/utils.dart:589]: Support is here: https://flutter-review.googlesource.com/c/recipes/+/31940

File: TESTOWNERS [hunk 3]
```diff
@@ -242,6 +242,9 @@
 /dev/integration_tests/android_embedding_v2_smoke_test @blasten @flutter/android
 /dev/integration_tests/release_smoke_test @blasten @flutter/android
 
+## OpenPay tests
+/dev/devicelab/bin/tasks/openpay_benchmarks__scroll_perf.dart @iskakaushik @flutter/engine
```
Comment (keyonghan) [inline TESTOWNERS:246]: The bot is distinguishing based on platform. This needs to be under linux/android section?

File: .ci.yaml [hunk 4]
```diff
@@ -2345,6 +2345,18 @@ targets:
       task_name: android_lifecycles_test
     scheduler: luci
 
+  - name: Linux_android openpay_benchmarks__scroll_perf
```
Comment (keyonghan) [inline .ci.yaml:2348]: We need a platform section for `linux_samsung_a02` similar to `s10`: https://github.com/flutter/flutter/blob/master/.ci.yaml#L42, but with `device_type: "SM-A025V"`.

And then rename the builder name to `Linux_samsung_a02 openpay_benchmarks__scroll_perf`.

File: .ci.yaml [hunk 5]
```diff
@@ -2345,6 +2345,18 @@ targets:
...
+    bringup: true
+    recipe: devicelab/devicelab_drone
+    presubmit: false
+    timeout: 60
+    properties:
+      openpay: "true"
+      tags: >
+        ["devicelab", "android", "linux", "openpay"]
```
Comment (keyonghan) [inline .ci.yaml:2356]: With the above comment, let's also add two more tags `samsung` and `a02`.

Comment (iskakaushik) [reply .ci.yaml]: Done!

File: .ci.yaml (hunk 4 above)
Comment (iskakaushik) [reply .ci.yaml]: Done!

File: TESTOWNERS (hunk 3 above)
Comment (iskakaushik) [reply TESTOWNERS]: Done!

**Review by keyonghan — APPROVED** [review]
LGTM

File: .ci.yaml [hunk 6]
```diff
@@ -2251,6 +2261,18 @@ targets:
...
+    bringup: true
+    recipe: devicelab/devicelab_drone
+    presubmit: false
+    timeout: 60
+    properties:
+      openpay: "true"
+      tags: >
+        ["devicelab", "android", "linux", "openpay", "samsung", "a02"]
+      task_name: openpay_benchmarks__scroll_perf
+    scheduler: luci
```
Comment (keyonghan) [inline .ci.yaml:2274]: Oops, sorry I missed this line. Now everything is via cocoon scheduling, could you remove this line?

Comment (iskakaushik) [reply .ci.yaml:2274]: https://github.com/flutter/flutter/pull/108126

**Pull request merged**